TShockAPI/Commands.cs

```diff
@@ -5424,51 +5491,60 @@ private static void Rocket(CommandArgs args)
 				{
 					ply.TPlayer.velocity.Y = -50;
 					TSPlayer.All.SendData(PacketTypes.PlayerUpdate, "", ply.Index);
-					args.Player.SendSuccessMessage("Rocketed {0}.", ply.Name);
+					if (args.Silent)
+						args.Player.SendSuccessMessage($"You rocketed {ply.Name}.");
+					else
+						// Should this be TSPlayer.All or plr?
+						TSPlayer.All.SendInfoMessage($"{args.Player.Name} has launched {ply.Name} into orbit.");
 				}
+				else if (!Main.ServerSideCharacter)
+					args.Player.SendErrorMessage("Server Side Character must be enabled to use this command!");
+				else if (!ply.IsLoggedIn)
+					args.Player.SendErrorMessage($"You cannot rocket {ply.Name} because {(ply.TPlayer.Male ? "he" : "she")} is not logged in.");
 				else
-				{
 					args.Player.SendErrorMessage("Failed to rocket player: Not logged in or not SSC mode.");
-				}
 			}
 		}
 
 		private static void FireWork(CommandArgs args)
 		{
 			if (args.Parameters.Count < 1)
 			{
-				args.Player.SendErrorMessage("Invalid syntax! Proper syntax: {0}firework <player> [red|green|blue|yellow]", Specifier);
+				args.Player.SendErrorMessage($"Invalid syntax! Proper syntax: {Specifier}firework <player> [red|green|blue|yellow]");
 				return;
 			}
 			var players = TSPlayer.FindByNameOrID(args.Parameters[0]);
 			if (players.Count == 0)
-				args.Player.SendErrorMessage("Invalid player!");
+				args.Player.SendErrorMessage($"{players} is not a valid player!");
```

Comment:
```suggestion
				args.Player.SendErrorMessage($"{args.Parameters[0]} is not a valid player!");
```
`players` will be an empty list here